Sources/Noora/Noora.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import Mockable
+
+@Mockable
+public protocol Noorable {
+    func singleChoicePrompt<T: CaseIterable & CustomStringConvertible & Equatable>(
+        question: String,
+        options: T.Type,
+        theme: NooraTheme
+    )
+
+    func singleChoicePrompt<T: CaseIterable & CustomStringConvertible & Equatable>(
+        title: String?,
+        question: String,
+        description: String?,
+        options: T.Type,
```

Comment:
We can derive the option directly from the generic.

If we do that, we enable the following syntax:
```swift
let option: MyOption = Noora().singleChoicePrompt(question: "What's it gonna be?")
// Instead of the current
let option = Noora().singleChoicePrompt(question: "What's it gonna be?", options: MyOption.self)
```